main/src/ca/uwaterloo/flix/util/FileOps.scala

```diff
@@ -27,6 +28,24 @@ import scala.util.Using
 
 object FileOps {
 
+  /**
+    * Deletes `file` if it exists. Wraps any error `e` in `Result.Err(e)`.
+    *
+    * @param file the file to delete
+    * @return `OK(())` if `file` was successfully deleted, `Err(e)` otherwise
+    */
+  def delete(file: Path): Result[Unit, Exception] = {
+    try {
+      if (Files.deleteIfExists(file)) {
```

Comment:
Note that this can throw DirectoryNotEmpty-- which can happen if someone is writing to it concurrently. Is that something to take into account.

Reply:
Yes but that seems like an issue always, right? You can never prevent other processes from writing to a specific dir.